crypto/secp256k1/secp256.go

```diff
@@ -86,6 +87,22 @@ func Sign(msg []byte, seckey []byte) ([]byte, error) {
 	return sig, nil
 }
 
+// SharedSecret computes the hashed ECDH shared secret.
+// pubkey must be an EC public key
+// seckey must be an EC private key
+func SharedSecret(pubkey []byte, seckey []byte) ([]byte, error) {
```

Comment:
I guess this should be added to the wallet-api, to be implemented by `keystore`, `externalsigner` and hardware wallets?